RecoEgamma/PhotonIdentification/src/PhotonIsolationCalculator.cc

```diff
@@ -430,82 +353,62 @@ void PhotonIsolationCalculator::calculate(const reco::Photon* pho,
-                                                     2);
-  phoisolR1.hcalDepth2TowerSumEt = HcalDepth2TowerIsoA;
-
-  double HcalDepth2TowerIsoB = calculateHcalTowerIso(pho,
-                                                     e,
-                                                     es,
-                                                     photonHcalDepth2TowerConeOuterRadiusB_,
-                                                     photonHcalDepth2TowerConeInnerRadiusB_,
-                                                     photonHcalDepth2TowerThreshEB_,
-                                                     2);
-  phoisolR2.hcalDepth2TowerSumEt = HcalDepth2TowerIsoB;
-
-  // New Hcal isolation based on the new H/E definition (towers behind the BCs in the SC are used to evaluated H)
-  double HcalTowerBcIsoA =
-      calculateHcalTowerIso(pho, e, es, photonHcalTowerConeOuterRadiusA_, photonHcalTowerThreshEA_, -1);
-  phoisolR1.hcalTowerSumEtBc = HcalTowerBcIsoA;
-
-  double HcalTowerBcIsoB =
-      calculateHcalTowerIso(pho, e, es, photonHcalTowerConeOuterRadiusB_, photonHcalTowerThreshEB_, -1);
-  phoisolR2.hcalTowerSumEtBc = HcalTowerBcIsoB;
-
-  //// Hcal depth1
-
-  double HcalDepth1TowerBcIsoA =
-      calculateHcalTowerIso(pho, e, es, photonHcalDepth1TowerConeOuterRadiusA_, photonHcalDepth1TowerThreshEA_, 1);
-  phoisolR1.hcalDepth1TowerSumEtBc = HcalDepth1TowerBcIsoA;
-
-  double HcalDepth1TowerBcIsoB =
-      calculateHcalTowerIso(pho, e, es, photonHcalDepth1TowerConeOuterRadiusB_, photonHcalDepth1TowerThreshEB_, 1);
-  phoisolR2.hcalDepth1TowerSumEtBc = HcalDepth1TowerBcIsoB;
-
-  //// Hcal depth2
-
-  double HcalDepth2TowerBcIsoA =
-      calculateHcalTowerIso(pho, e, es, photonHcalDepth2TowerConeOuterRadiusA_, photonHcalDepth2TowerThreshEA_, 2);
-  phoisolR1.hcalDepth2TowerSumEtBc = HcalDepth2TowerBcIsoA;
-
-  double HcalDepth2TowerBcIsoB =
-      calculateHcalTowerIso(pho, e, es, photonHcalDepth2TowerConeOuterRadiusB_, photonHcalDepth2TowerThreshEB_, 2);
-  phoisolR2.hcalDepth2TowerSumEtBc = HcalDepth2TowerBcIsoB;
+  if (not hbheRecHitsTag_.isUninitialized()) {
+    auto const& hbheRecHits = e.get(hbheRecHitsTag_);
+
+    for (size_t id = 0; id < phoisolR1.hcalRecHitSumEt.size(); ++id) {
+      phoisolR1.hcalRecHitSumEt[id] =
+          calculateHcalRecHitIso<false>(pho,
+                                        caloGeometry,
+                                        *hcalTopology,
+                                        *hcalChannelQuality,
+                                        *hcalSevLvlComputer,
+                                        towerMap,
+                                        hbheRecHits,
+                                        detector == EcalBarrel ? hcalIsoOuterRadAEB_[id] : hcalIsoOuterRadAEE_[id],
+                                        detector == EcalBarrel ? hcalIsoInnerRadAEB_[id] : hcalIsoInnerRadAEE_[id],
+                                        id + 1);
```

Comment:
can this be simplified with `std::bind`

Reply:
sure. gimme some mins.

Reply:
just to be clear, I was referring to repeated calls here and below